sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MarkAssemblyWithAttributeBase.cs

```diff
@@ -39,7 +39,7 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                             var requiredAttributeFound = cc.Compilation.Assembly.GetAttributes(AttributeToFind).Any();
                             if (!requiredAttributeFound)
                             {
-                                cc.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0], null));
+                                cc.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0], null, cc.Compilation.AssemblyName));
```

Comment:
According to [doc](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.compilation.assemblyname?view=roslyn-dotnet), `Compilation.AssemblyName` can be also null. 

Could we handle also this case?

Reply:
I don't think I can force the assemblyName to null in our test framework. If we can, I don't know how, and I'm not sure it's worth investing time into this, since this parameter is used in a formatting string, so even if it is null, the formatted string will show empty string `''` instead of the assembly name, which IMO is acceptable given the rare case (if any) it should happen